File: ejercicios/c++/ejercicio081.cpp
#include <iostream>
#include <ctime>
#include <cstdlib>
using namespace std;

int main(){
	int ar[10];
	int max, min, cerca, promedio, acum = 0;
	srand(time(0));
	for(int i = 0; i < 10; i++){
		ar[i] = (rand() % 100);
		acum = acum + ar[i];
	}
	promedio = acum / 10;
	max = ar[0];
	min = ar[0];
	cerca = ar[0];
	for(int i = 0; i < 10; i++){
		if(ar[i] > max){
			max = ar[i];
		}
		if(ar[i] < min){
			min = ar[i];
		}
		if(abs(promedio - ar[i]) < abs(promedio - cerca)){
			cerca = ar[i];
		}
	}
	
	cout << "Elementos del arreglo: " << endl;
	for(int i=0;i<10;i++)
		cout << ar[i] << "  "; 
	
	cout << endl << " el menor es : " << min << endl;
	cout << " el mayor es : " << max << endl;
	cout << " el promedio es: " << promedio << endl;
	cout << " el mas cercano a "<< promedio << " es : " << cerca << endl;
	return 0;
}

File: ejercicios/python/ejercicio081.py
import random
import array

ar = []
#ar es una lista
acum = 0

for i in range(0,10):
    ar.append(random.randint(0, 100))
    acum += ar[i]

promedio = acum // 10
maximo = ar[0]
minimo = ar[0]
cerca = ar[0]

for i in range(0,10):
    if ar[i] > maximo:
        maximo = ar[i]
    if ar[i] < minimo:
        minimo = ar[i]
    if abs(promedio - ar[i]) < abs(promedio - cerca):
        cerca = ar[i]

print("Elementos del arreglo:\n" + str(ar))

print( " el menor es : " + str(minimo) )
print( " el mayor es : " + str(maximo) )
print( " el promedio es: " + str(promedio) )
print( " el mas cercano a "+ str(promedio) + " es : " + str(cerca) )
